### ISSUE TITLE
Set content type for CSV so that browsers will attempt to download instead opening in the browser

### ISSUE THREAD

**Issue by fgregg** [simonw/datasette#1549]
Right now, if the user clicks on the CSV related to a table or a query, the response header for the content type is 

"content-type: text/plain; charset=utf-8"

Most browsers will try to open a file with this content-type in the browser. 

This is not what most people want to do, and lots of folks don't know that if they want to download the CSV and open it in the a spreadsheet program they next need to save the page through their browser.

It would be great if the response header could be something like 

```
'Content-type: text/csv');
'Content-disposition: attachment;filename=MyVerySpecial.csv');
```

which would lead browsers to open a download dialog.

**Comment by simonw** [comment]
That's what this option does:

![EAB1B9E8-38E9-4C6D-8854-BD1935F163D9](https://user-images.githubusercontent.com/9599/145688531-668bafa1-e287-4bbd-84d6-157241fb1f68.jpeg)

The usability of this is pretty terrible though (including "stream all rows" - how are people meant to understand what that does?) so it can definitely do with some rethinking.

**Comment by fgregg** [comment]
that option is not available on [custom queries](https://labordata.bunkum.us/odpr-962a140?sql=with+local_union_filings+as+%28%0D%0A++select+*+from+lm_data+%0D%0A++where%0D%0A++++yr_covered+%3E+cast%28strftime%28%27%25Y%27%2C+%27now%27%2C+%27-5+years%27%29+as+int%29%0D%0A++++and+desig_name+%3D+%27LU%27%0D%0A++order+by+yr_covered+desc%0D%0A%29%2C%0D%0Amost_recent_filing+as+%28%0D%0A++select%0D%0A++++*%0D%0A++from+local_union_filings%0D%0A++group+by%0D%0A++++f_num%0D%0A%29%0D%0Aselect%0D%0A++*%0D%0Afrom%0D%0A++most_recent_filing%0D%0Awhere%0D%0A++next_election+%3E%3D+strftime%28%27%25Y-%25m%27%2C+%27now%27%29%0D%0A++and+next_election+%3C+strftime%28%27%25Y-%25m%27%2C+%27now%27%2C+%27%2B1+year%27%29%0D%0Aorder+by%0D%0A++members+desc%3B).

**Comment by simonw** [comment]
Good call! I'm doing a refactor #1518 right now which will hopefully bring the functionality of those two much closer - I'll make a note to consider this there too.

**Comment by simonw** [comment]
Also relevant:
- #1062

**Comment by fgregg** [comment]
another way to get the behavior of downloading the file is to use the download attribute of the anchor tag

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download
